lib/chef/knife/ssh.rb

```diff
@@ -204,6 +207,12 @@ def search_nodes
         list
       end
 
+      def fuzzify_query
+        if @query !~ /:/
+          @query = "tags:*#{@query}* OR roles:*#{@query}* OR fqdn:*#{@query}* OR addresses:*#{@query}*"
```

Comment:
Should this be "weaker"? Possibilities:

1. Remove asterisks for both sides.
2. fqdn only? fqdn + name?
3. Keep as is, but have a confirmation prompt for what fqdn we're running  on. (Can be explicitly skipped with `--yes` flag.)

cc: @lamont-granquist